tests/test_other.py

```diff
@@ -1524,6 +1524,53 @@ def test(link_flags, lib_suffix):
     test(['-lfile'], '') # -l, auto detection from library path
     test([self.in_dir('libdir', 'libfile.so.3.1.4.1.5.9')], '.3.1.4.1.5.9') # handle libX.so.1.2.3 as well
 
+  def test_dynamic_link_module_heap_in_pthread(self):
+    # Test that a side module uses the same heap for global objects across all threads
```

Comment:
The term heap is a littel misleading here I think.   How about refering to it as he static data region (or segment) of of the side module?

Perhaps this test can be called `test_dynamic_link_pthread_static_data`?